### PR TITLE
feat(icon): add bun.lock for bun

### PR DESCRIPTION
**Pull request by nguyenyou**
🎉 We now have bun lock as text: https://bun.sh/blog/bun-lock-text-lockfile

**Changes proposed:**

- [x] Add
- [ ] Delete
- [ ] Fix
- [ ] Prepare

### CODE REVIEW

**Comment by jakeboone02** [comment]
I added `'bun.lock'` in #3636 but it didn't seem to work:

image

I think your update may work but you might need to add `filenamesGlob: ['bun']` or something similar as well. I'm not 100% sure how the file name matching algorithm works for this extension.

**Comment by nguyenyou** [comment]
@jakeboone02 
Looks like we need to add ` filename: true`

File: src/iconsManifest/supportedExtensions.ts [hunk 1]
```diff
@@ -837,7 +837,8 @@ export const extensions: IFileCollection = {
     },
     {
       icon: 'bun',
-      extensions: ['bun', 'lockb', 'bun.lock'],
+      extensions: ['bun', 'bun.lockb', 'bun.lock'],
```
Comment (remcohaszing) [inline src/iconsManifest/supportedExtensions.ts:840]: I don’t think there’s a file named `bun`, is there? I do see there are [`bunfig.toml` and `.bundig.toml`](https://bun.sh/docs/runtime/bunfig). You might as well add those while at it.

```suggestion
      extensions: ['.bunfig.toml', 'bunfig.toml', 'bun.lockb', 'bun.lock'],
```

Also Bun for Visual Studio Code adds the [`bun.lockb`](https://github.com/oven-sh/bun/blob/b5b51004e8de80dcd43849e4da1d2623ee3d3008/packages/bun-vscode/package.json#L288) language ID. You could register the icon for that too, so it shows up in the language picker.

**Review by remcohaszing — APPROVED** [review]
Thanks! I approved because the change is good as-is, but I refrained from merging for now, because I think it can be improved even further.

File: src/iconsManifest/supportedExtensions.ts (hunk 1 above)
Comment (nguyenyou) [reply src/iconsManifest/supportedExtensions.ts]: I'm not sure about how to config the language ID as you mentioned, please help!

**Review by remcohaszing — DISMISSED** [review]

File: src/iconsManifest/supportedExtensions.ts (hunk 1 above)
Comment (nguyenyou) [reply src/iconsManifest/supportedExtensions.ts]: Done 👍

**Review by remcohaszing — APPROVED** [review]

**Pull request merged**